File: 03 C C++ Python R and Mojo review/03-2 Python Review/01 Pointers/01.py
# In Python, variables are references to objects, so no explicit pointers are required

x = 10
ptr = x  # 'ptr' references the same value as 'x'

# Python doesn't expose memory addresses directly, but we can use id() to get the memory address.
print(f"Address of x: {id(ptr)}")  # Prints the memory address (not exactly like C)
print(f"Value of x: {ptr}")  # Prints the value of x

File: 03 C C++ Python R and Mojo review/03-3 R Language Review/01 Pointers/01.R
x <- 10
ptr <- x  # In R, variables are references

# R doesn't have a direct way to show memory addresses, but you can use the "address" of an object 
# using the "pryr" package for low-level memory inspection
library(pryr)
cat("Address of x: ", address(ptr), "\n")  # Address function from pryr package
cat("Value of x: ", ptr, "\n")